creusot-contracts/src/logic/int.rs

```diff
@@ -6,7 +6,7 @@ use creusot_contracts_proc::*;
 use std::ops::*;
 
 #[rustc_diagnostic_item = "creusot_int"]
-pub struct Int;
+pub struct Int(*mut ());
```

Comment:
Yea, agreed on the addition of an `Opaque` type, I just got lazy here but made the same reflection to myself as this will be a systemic pattern for built in types.

As for the translation in why3 `*mut T` becomes `opaque_ptr T` where `opaque_ptr` is a type declaration (not definition).